android/src/main/java/com/pw/droplet/braintree/Braintree.java

```diff
@@ -1,301 +1,316 @@
+import com.braintreepayments.api.PayPalVaultRequest;
+import com.braintreepayments.api.PaymentMethodNonce;
+import com.braintreepayments.api.UserCanceledException;
+import com.braintreepayments.api.VenmoAccountNonce;
+import com.braintreepayments.api.VenmoClient;
+import com.braintreepayments.api.VenmoPaymentMethodUsage;
+import com.braintreepayments.api.VenmoRequest;
 import com.facebook.react.bridge.ActivityEventListener;
-import com.facebook.react.bridge.BaseActivityEventListener;
-import com.facebook.react.bridge.Callback;
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.LifecycleEventListener;
+import com.facebook.react.bridge.Promise;
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.ReactContextBaseJavaModule;
 import com.facebook.react.bridge.ReactMethod;
 import com.facebook.react.bridge.ReadableMap;
 import com.facebook.react.bridge.WritableMap;
-import com.facebook.react.bridge.WritableNativeMap;
-import com.google.android.gms.wallet.PaymentData;
 import com.google.android.gms.wallet.TransactionInfo;
 import com.google.android.gms.wallet.WalletConstants;
-import com.google.gson.Gson;
 
-import java.util.HashMap;
-import java.util.Map;
 
-public class Braintree extends ReactContextBaseJavaModule {
-    private String token;
-    private Callback deviceDataCallback;
-    private Boolean collectDeviceData = false;
-    private Callback successCallback;
-    private Callback errorCallback;
+public class Braintree extends ReactContextBaseJavaModule
+        implements ActivityEventListener, LifecycleEventListener {
 
-    private BraintreeFragment mBraintreeFragment;
-
-    public Braintree(ReactApplicationContext reactContext) {
-        super(reactContext);
-        reactContext.addActivityEventListener(activityEventListener);
-    }
+    private final Context mContext;
+    private FragmentActivity mCurrentActivity;
+    private Promise mPromise;
+    private String mDeviceData;
+    private String mToken;
+    private BraintreeClient mBraintreeClient;
+    private PayPalClient mPayPalClient;
+    private GooglePayClient mGooglePayClient;
+    private VenmoClient mVenmoClient;
 
     @Override
     public String getName() {
         return "Braintree";
     }
 
-    public String getToken() {
-        return this.token;
+    public Braintree(ReactApplicationContext reactContext) {
+        super(reactContext);
+
+        mContext = reactContext;
+
+        reactContext.addLifecycleEventListener(this);
+        reactContext.addActivityEventListener(this);
     }
 
-    public void setToken(String token) {
-        this.token = token;
+    @Override
+    public void onActivityResult(Activity activity, int requestCode, int resultCode, Intent intent) {
+        switch (requestCode) {
+            case BraintreeRequestCodes.GOOGLE_PAY:
+                if (mGooglePayClient != null) {
+                    mGooglePayClient.onActivityResult(
```

Comment:
onActivityResult depreciated, need to use GooglePayListener

![image](https://github.com/StronkMan/react-native-braintree-no-ui/assets/165301863/0e5f9ecb-dab5-4f22-9cda-c77e088e48bf)